src/main/java/uk/gov/companieshouse/model/dto/email/EmailDocument.java

```diff
@@ -0,0 +1,67 @@
+package uk.gov.companieshouse.model.dto.email;
+
+public class EmailDocument<T> {
```

Comment:
I wonder if this should be `T extends EmailData` instead

Reply:
I think this is fine, I don't think we need to repeat it here. The type that we substitute in should extend EmailData anyway, e.g. SuccessfulPaymentEmailData extends EmailData